### PR TITLE
Stats frontend

### PR DESCRIPTION
**Pull request by hskowron**

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -16,20 +16,39 @@ import classes from "./styles.module.scss";
...
 
 export default function App() {
   const [userSocket] = useState<UserSocket>(new UserSocket());
+  const [channelSocket] = useState<ChannelSocket>(new ChannelSocket());
   const [image, setImage] = useImageStore((state) => [
     state.image,
     state.setImage,
   ]);
   const imageUrl = image ? URL.createObjectURL(image) : "";
+  const [token, setToken] = useState<string>("");
```
Comment (williamollio) [inline ui/src/App.tsx:32]: const [token] = useState<string | null>(localStorage.getItem(Cookie.TOKEN));

File: ui/src/components/chat/ChannelInfoDialog.tsx [hunk 2]
```diff
@@ -93,24 +96,32 @@ export default function ChannelInfoDialog({
   };
 
   useEffect(() => {
-    ["userConnected, userDisconnected"].forEach((element) => {
-      userSocket.socket?.on(element, () => {
+    // if (userSocket.socket.connected) {
```
Comment (williamollio) [inline ui/src/components/chat/ChannelInfoDialog.tsx:99]: did you forget it ?

File: ui/src/interfaces/stats.interface.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+export interface leaderBoardEntry {
+  id: string;
+  name: string;
+  rank: number;
+  rating: number;
+//   wins: number;
```
Comment (williamollio) [inline ui/src/interfaces/stats.interface.ts:6]: it's upcoming right ?

**Review by williamollio — APPROVED** [review]

File: ui/src/interfaces/stats.interface.ts (hunk 3 above)
Comment (hskowron) [reply ui/src/interfaces/stats.interface.ts:6]: Yes I asked Mouad to add wins and losses to the getLeaderboard request. I would have to make individual requests for every player in the leaderboard which seems like a little overkill.

File: ui/src/components/chat/ChannelInfoDialog.tsx (hunk 2 above)
Comment (hskowron) [reply ui/src/components/chat/ChannelInfoDialog.tsx:99]: Yes ... damn

**Pull request merged**